lib/perl/Genome/Model/ClinSeq/Command/UpdateAnalysis.pm

```diff
@@ -820,12 +839,42 @@ sub get_trsn{
 
   my %trsns;
   my $trsn_ref;
-
-  foreach my $instrument_data (@instrument_data){
-    my $trsn = $instrument_data->target_region_set_name;
-    if ($trsn){
-      $trsns{$trsn}=1;
-      $trsn_ref = $trsn;
+  
+  #When using validation-as-exome and my-trsn options, use user-defined target region set name (1=exome or 2=validation)
+  if ($self->validation_as_exome && $self->my_trsn) {
+    my $user_trsn = $self->my_trsn;
+    
+    #Check if a valid option for my-trsn and use or warn user and exit
+    if ($user_trsn == 1){
```

Comment:
Is there a reason you're using numbers here? Depending on if you need the numeric values or not, you could let UR handle this validation for you.

Instead of 
````perl
        my_trsn => {
              is => 'Number',
              doc => 'If using validation-as-exome option, option to specify what instrument data to derive the target-set-region-name and region-of-interest from. 1=exome or 2=validation', 
        }
````

You could do something like:

````perl
    my_trsn => {
              is => 'Text',
             doc => 'If using validation-as-exome option, option to specify what instrument data to derive the target-set-region-name and region-of-interest from.', 
            valid_values => ['exome', 'validation']
        },
````

Which will constrain the values to either `exome` or `validation` for you. Then you could change your `if` block to be a little simpler:

````perl

#no need to double check that `my_trsn` is set here as you're checking it near line 260
if ($self->validation_as_exome ) {
    my $user_trsn = $self->my_trsn;
    $self->status_message("Using $user_trsn target region specified by my-trsn");
}
````